rest_framework-stubs/decorators.pyi

```diff
@@ -29,43 +30,62 @@ class MethodMapper(dict):
     def options(self, func: _View) -> _View: ...
     def trace(self, func: _View) -> _View: ...
 
-_LOWER_CASE_HTTP_VERBS: TypeAlias = Sequence[
-    Literal[
-        "get",
-        "post",
-        "delete",
-        "put",
-        "patch",
-        "trace",
-        "head",
-        "options",
-    ]
-]
+if sys.version_info >= (3, 11):
+    from http import HTTPMethod
 
-_MIXED_CASE_HTTP_VERBS: TypeAlias = Sequence[
-    Literal[
-        "GET",
-        "POST",
-        "DELETE",
-        "PUT",
-        "PATCH",
-        "TRACE",
-        "HEAD",
-        "OPTIONS",
-        "get",
-        "post",
-        "delete",
-        "put",
-        "patch",
-        "trace",
-        "head",
-        "options",
+    _MIXED_CASE_HTTP_VERBS: TypeAlias = Sequence[
+        Literal[
+            "GET",
+            "POST",
+            "DELETE",
+            "PUT",
+            "PATCH",
+            "TRACE",
+            "HEAD",
+            "OPTIONS",
+            "get",
+            "post",
+            "delete",
+            "put",
+            "patch",
+            "trace",
+            "head",
+            "options",
+            HTTPMethod.GET,
+            HTTPMethod.POST,
+            HTTPMethod.DELETE,
+            HTTPMethod.PUT,
+            HTTPMethod.PATCH,
+            HTTPMethod.TRACE,
+            HTTPMethod.HEAD,
+            HTTPMethod.OPTIONS,
```

Comment:
Sure, let's try to go this way, I just really know that `Literal[HTTPMethod.GET]` is working when you specify it without variable, but if you try to use variable instead, like:
```python
my_var: HTTPMethod
my_var = HTTMethod.GET if ... else HTTPMethod.POST

@action(http_methods=[my_var],...)
def my_view(...) -> Respones: ...
```
it will not work as it will complain that your `my_var` is not Literal

If we go your approach, that this solution will also work, I suspect, therefore it is even better. Let's try it out

Reply:
Used the union with `HTTPMethod` enum as suggested and bot tests for `LIteral` and `non-Literal` passing